torch/fx/experimental/sym_node.py

```diff
@@ -170,6 +170,18 @@ def _value_hash(self) -> int:
 
     @property
     def expr(self):
+        result = self.expr_allow_custom_add
+        # We make sure that CustomAdd is not leaking outside the construction loop.
+        # This is needed to allow Add(a, b) == CustomAdd(a, b) to be true.
+        if isinstance(result, torch.utils._sympy.functions.CustomAdd):
+            import sympy
+
+            return sympy.Add(*result._args, evaluate=False)
```

Comment:
this is the source of the 2% regression when N=100
when N=1000 we have 60% win though!

Reply:
actually this is not acceptable it reduce savings by 20s, looking at avoiding this.